IBX-3135: Updated missing components (#1813)
ezsystems/developer-documentation: docs/administration/back_office/back_office_elements/reusable_components.md

} only %}
```

## Details

The details component consists of the following blocks:

- `details_header`

- `details_items`

Variables:

|Name|Type|Values|
|----|----|-----------|

> |`headline`|string|`@ibexadesign/ui/component/table/table_header.html.twig`|

Reviewer: ```suggestion
|`headline`|string|
```
?

> |`headline_items`|array|`@ibexadesign/ui/component/table/table_header.html.twig`|

Reviewer: ```suggestion
|`headline_items`|array|
```
?

|`view_mode`|string|`vertical`, default set to `''`|
|`items`|array|`label`, `content_raw`, `content`|

`headline` and `headline_items` are variables for `@ibexadesign/ui/component/table/table_header.html.twig`

## Modal

The modal component consist of the following blocks:

```html+twig
{% block dialog %}
	{% block content_before %}
	{% block content %}
		{% block header %}
		{% block subheader %}
		{% block body %}
		{% block footer %}
	{% block content_after %}
```
Variables:

|Name|Type|Values|
|----|----|-----------|
|`size`|string|`small`, `large`, `extra-large`, default set to: `''`|

> |`subtitle`|string|default set to `false`|

Reviewer 2: It is not by default set to `false`. We have `subtitle|default(false)` in one condition but this is not an assignment of a default value. We can see in the template, that below we have a check whether `subtitle` is defined:
```
{% if subtitle is defined %}
    <div class="ibexa-modal__subheader">
        {{ subtitle }}
    </div>
{% endif %}
```
so I would state that there is no default value and also add that when it is not defined subheader is not rendered.

|`no_header`|boolean|default set to `false`|
|`no_header_border`|boolean|default set to `false`|
|`class`|string|default `''`|

> |`tabindex`|string|default `-1`|

Reviewer 2: There is no `tabindex` parameter on the component itself; it is part of `attr`:
`tabindex: attr.tabindex|default('-1'),`

> |`role`|string|default `dialog`|

Reviewer 2: same as `tabindex`

|`id`|string||
|`has_static_backdrop`|boolean|default set to `false`|

> `attr`

Reviewer 2: If we want to be precise, it would probably be good to state that `attr` and other `attr_*` hold all HTML attributes rendered on their respective elements, but some are defaulted or modified.

But maybe it is stated in another part of the documentation. :)

} only %}
```

## Details

The details component consists of the following blocks:

- `details_header`

> 

Reviewer 3: ```suggestion
```

- `details_items`

Variables:

|Name|Type|Values|
|----|----|-----------|
|`headline` (optional)|string|if not defined, `details_header` is empty|
|`headline_items`|array|
|`view_mode`|string|`vertical`, default set to `''`|
|`items`|array<hash>|{`label`, `content_raw`, `content`}|

> If `headline` is passed, the `table_header` is loaded to `details_header` and then it is possible to pass `headline_items` variable.

Reviewer 3: ```suggestion
If `headline` is passed, a `table_header` element is loaded in `details_header` and then it is possible to pass a `headline_items` variable.
```
pass to which element, `details_header`?

> `headline` and `headline_items` are variables for `@ibexadesign/ui/component/table/table_header.html.twig`

Reviewer 3: ```suggestion
`headline` and `headline_items` are variables used in `@ibexadesign/ui/component/table/table_header.html.twig`
```
?

## Modal

> The modal component consist of the following blocks:

Reviewer 3: ```suggestion
The modal component consists of the following blocks:
```

```html+twig
{% block dialog %}

> 	{% block content_before %}

Reviewer 3: Cosmetic: too big indents. This should not matter in functionality, but doesn't look nice.

	{% block content %}
		{% block header %}
		{% block subheader %}
		{% block body %}
		{% block footer %}
	{% block content_after %}
```
Variables:

|Name|Type|Values|
|----|----|-----------|
|`size`|string|`small`, `large`, `extra-large`, default set to: `''`|
|`subtitle`|string|no default value, if not defined, the `subheader` is not rendered|
|`no_header`|boolean|default set to `false`|
|`no_header_border`|boolean|default set to `false`|
|`class`|string|default `''`|

> 
> 

Reviewer 3: ```suggestion
```
This breaks table rendering (but I can't add a suggestion for both lines)

|`id`|string||
|`has_static_backdrop`|boolean|default set to `false`|

`attr` and other `attr_*` hold all HTML attributes rendered on their respective elements.

> `attr`

Reviewer 3: Set this and below as heading?

|Name|Type|Values|
|----|----|-----------|
|`class`|string|default `''`|
|`role`|string|default `dialog`|
|`tabindex`|string|default `-1`|

`attr_dialog`

|Name|Type|Values|
|----|----|-----------|
|`class`|string|default set to `''`|
|`role`|string|default set to `document`|

`attr_content`

|Name|Type|Values|
|----|----|-----------|
|`class`|string|default set to `''`|

`attr_title`

|Name|Type|Values|
|----|----|-----------|
|`class`|string|default set to `''`|

`attr_close_btn`

|Name|Type|Values|
|----|----|-----------|

> |`class`||default set to `''`|

Reviewer 3: types missing here?
